scripts/coverage_growth.py

```
@@ -21,224 +23,292 @@
+    missing = _line_set(payload.get("missing_lines"), f"{label} missing_lines")
+    if executed & missing:
+        raise CoverageRatchetError(f"{label} coverage line sets overlap")
+    summary = payload.get("summary", {})
+    if not isinstance(summary, dict):
+        raise CoverageRatchetError(f"{label} line summary is malformed")
+    if summary.get("num_statements") and not executed and not missing:
+        raise CoverageRatchetError(f"{label} has statements but no executable line records")
+    if summary:
+        _validate_counts(summary, "covered_lines", "num_statements", label)
+        if summary["num_statements"] != len(executed | missing) or summary["covered_lines"] != len(executed):
+            raise CoverageRatchetError(f"{label} executable lines differ from summary")
```

Ruff (0.16.3)

[warning] Avoid specifying long messages outside the exception class

(TRY003)